src/meta/src/hummock/manager/mod.rs

```diff
@@ -1160,85 +1266,127 @@ where
+            if !new_hummock_version.levels.contains_key(group_id) {
                 new_hummock_version
                     .levels
                     .try_insert(
-                        group_id,
-                        <Levels as HummockLevelsExt>::build_initial_levels(&compaction_config),
+                        *group_id,
+                        <Levels as HummockLevelsExt>::build_initial_levels(compaction_config),
                     )
                     .unwrap();
+                new_groups.push(*group_id);
+                need_independent_commit = true;
                 let level_deltas = &mut new_version_delta
                     .level_deltas
-                    .entry(group_id)
+                    .entry(*group_id)
                     .or_default()
                     .level_deltas;
                 level_deltas.push(LevelDelta {
                     delta_type: Some(DeltaType::GroupConstruct(GroupConstruct {
-                        group_config: Some(compaction_config),
+                        group_config: Some(compaction_config.clone()),
                     })),
                 });
             }
         }
+
+        if !need_independent_commit {
+            return Ok((
+                new_version_delta.key,
+                new_version_delta.new_value,
+                new_hummock_version,
+            ));
+        }
+
+        new_version_delta.max_committed_epoch = new_hummock_version.max_committed_epoch;
+        commit_multi_var!(self, None, new_version_delta)?;
+        branched_ssts.commit();
+        versioning.current_version = new_hummock_version;
+
+        self.env
+            .notification_manager()
+            .notify_hummock_asynchronously(
+                Operation::Add,
+                Info::HummockVersionDeltas(risingwave_pb::hummock::HummockVersionDeltas {
+                    version_deltas: vec![versioning
+                        .hummock_version_deltas
+                        .last_key_value()
+                        .unwrap()
+                        .1
+                        .clone()],
+                }),
+            );
+
+        old_version = versioning.current_version.clone();
+        new_version_id = old_version.id + 1;
+        let mut new_version_delta1 = BTreeMapEntryTransaction::new_insert(
+            &mut versioning.hummock_version_deltas,
+            new_version_id,
+            HummockVersionDelta {
+                prev_id: old_version.id,
+                safe_epoch: old_version.safe_epoch,
+                trivial_move: false,
+                ..Default::default()
+            },
+        );
+
+        new_hummock_version = old_version;
+        new_version_delta1.id = new_version_id;
+        new_hummock_version.id = new_version_id;
+        Ok((
+            new_version_delta1.key,
+            new_version_delta1.new_value,
+            new_hummock_version,
+        ))
     }
 
     /// Caller should ensure `epoch` > `max_committed_epoch`
```

Comment:
Why do we need a new version here with no updates on version deltas and epoch?

Reply:
> Why do we need a new version here with no updates on version deltas and epoch?

If some input SSTs in `commit_epoch` are broken, we might fail to `commit_epoch`. In this situation, we want to finish `sync_group` at least.

Reply:
I still don't get it. If there are no level/group delta updates and epoch updates, what can we gain from adding a dummy version delta?

Reply:
> I still don't get it. If there are no level/group delta updates and epoch updates, what can we gain from adding a dummy version delta?

If Some SSTs are broken, we need to loop
`sync_group`, `commit_epoch` and then rollback both cause SST broken
`sync_group`, `commit_epoch` and then rollback both cause SST still broken
`sync_group`, `commit_epoch` and then rollback both cause SST broken again
To prevent this, we commit `sync_group` first. Thus we loop
`commit_epoch` and then rollback `commit_epoch` only